compiler/rustc_resolve/src/late.rs

```diff
@@ -2069,7 +2415,7 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
             {
                 diagnostics::signal_lifetime_shadowing(self.r.session, original, param.ident);
                 // Record lifetime res, so lowering knows there is something fishy.
-                self.record_lifetime_res(param.id, LifetimeRes::Error);
+                self.r.lifetimes_res_map.insert(param.id, LifetimeRes::Error);
```

Comment:
```suggestion
                self.record_lifetime_res(param.id, LifetimeRes::Error, LifetimeElisionCandidate::Ignore);
```
Or something like that.
`record_lifetime_res` checks for duplicate insertions after all, I think it's better to keep using it consistently.

Reply:
`record_lifetime_res` also records lifetimes as to be candidates for elision.  Generic parameters are not candidates for elision, so we should not use the same function.  We could introduce a different `lifetime_param_res` but that would probably be overkill.

Reply:
>`record_lifetime_res` also records lifetimes as to be candidates for elision. Generic parameters are not candidates for 

But it does not record if the resolution is `LifetimeRes::Error` (or `LifetimeRes::ElidedAnchor`), so `record_lifetime_res` can be used in at least those cases.

I'd probably use two functions - `record_lifetime_res` and `record_lifetime_res_and_also_add_a_lifetime_elision_candidate` (but with a shorter name).
The latter could also panic on `LifetimeRes::Error` and `LifetimeRes::ElidedAnchor` instead of doing nothing.